fbw-a32nx/src/systems/systems-host/systems/FWC/PseudoFWC.ts

```diff
@@ -1332,6 +1507,194 @@ export class PseudoFWC {
+      this.soundManager.enqueueSound('cavalryChargeOnce'); // On the A320, play first cav charge completely no matter what
+    }
+    if (this.autoPilotOffVoluntaryFirstCavalryChargeActivePulse.read()) {
+      this.soundManager.dequeueSound('cavalryChargeOnce');
+    }
+    if (!this.autoPilotOffVoluntaryMemory.read() && !this.autoPilotOffInvoluntaryMemory.read()) {
+      this.requestMasterWarningFromApOff = false;
+      this.soundManager.dequeueSound('cavalryChargeOnce');
+      this.soundManager.dequeueSound('cavalryChargeCont');
+    }
+
+    this.autoPilotInstinctiveDiscPressedPulse.write(false, deltaTime);
+
+    // approach capability downgrade. Debounce first, then suppress for a certain amount of time
+    // (to avoid multiple triple clicks, and a delay which is too long)
+    let fmgcApproachCapability = 0;
+    const getApproachCapability = (dw3: Arinc429WordData, dw4: Arinc429WordData): number => {
+      let appCap = 0;
+      const landModeActive = dw4.bitValueOr(14, false);
+      const landModeArmed = dw3.bitValueOr(20, false);
+      const land2Capacity = dw4.bitValueOr(23, false);
+      const land3FailPassiveCapacity = dw4.bitValueOr(24, false);
+      const land3FailOperationalCapacity = dw4.bitValueOr(25, false);
+
+      if (land2Capacity) {
+        appCap = 2;
+      } else if (land3FailPassiveCapacity) {
+        appCap = 3;
+      } else if (land3FailOperationalCapacity) {
+        appCap = 4;
+      } else if (landModeActive || landModeArmed) {
+        appCap = 1;
+      } else {
+        appCap = 0;
+      }
+      return appCap;
+    };
+
+    if (this.fmgc1DiscreteWord3.get().isNormalOperation() && this.fmgc1DiscreteWord4.get().isNormalOperation()) {
+      fmgcApproachCapability = getApproachCapability(this.fmgc1DiscreteWord3.get(), this.fmgc1DiscreteWord4.get());
+    } else if (this.fmgc2DiscreteWord3.get().isNormalOperation() && this.fmgc2DiscreteWord4.get().isNormalOperation()) {
+      fmgcApproachCapability = getApproachCapability(this.fmgc2DiscreteWord3.get(), this.fmgc2DiscreteWord4.get());
+    }
+
+    const capabilityDowngrade =
+      fmgcApproachCapability < this.fmgcApproachCapability.get() && fmgcApproachCapability > 0;
+    this.approachCapabilityDowngradeDebounce.write(
+      capabilityDowngrade && flightPhase167 && !this.approachCapabilityDowngradeSuppress.read(),
+      deltaTime,
+    );
+    this.approachCapabilityDowngradeDebouncePulse.write(this.approachCapabilityDowngradeDebounce.read(), deltaTime);
+    this.approachCapabilityDowngradeSuppress.write(this.approachCapabilityDowngradeDebouncePulse.read(), deltaTime);
+    // Capability downgrade after debounce --> triple click
+    if (this.approachCapabilityDowngradeDebouncePulse.read()) {
+      this.soundManager.enqueueSound('tripleClick');
+    }
+    this.fmgcApproachCapability.set(fmgcApproachCapability);
+
+    // A/THR OFF
+    const aThrEngaged = this.autoThrustStatus.get() === 2 || this.autoThrustMode.get() !== 0;
```

Comment:
I believe this is the cause of the autothrust off message triggering when going MCT or TOGA when previously engaged, as the status will go to 1 (engaged_armed). In addition to that, is the mode check also required?